llvm/include/llvm/Target/GlobalISel/Combine.td

```diff
@@ -1948,7 +1956,7 @@ def all_combines : GICombineGroup<[integer_reassoc_combines, trivial_combines,
     fsub_to_fneg, commute_constant_to_rhs, match_ands, match_ors,
     combine_concat_vector, match_addos,
     sext_trunc, zext_trunc, prefer_sign_combines, combine_shuffle_concat,
-    combine_use_vector_truncate, merge_combines]>;
+    combine_use_vector_truncate, merge_combines, combine_shuffle_undef_rhs]>;
```

Comment:
Can you create a new combine group, maybe `shuffle_combines`, and put this and the `combine_shuffle_concat` combine into it? Can throw in anything else you think fits the scope there too.